src/lib.rs

```diff
@@ -338,13 +324,17 @@ pub fn generate_flamegraph_for_workload(workload: Workload, opts: Options) -> an
         signal_hook::low_level::register(SIGINT, || {}).expect("cannot register signal handler")
     };
 
-    let use_sudo = opts.root.is_some();
+    let sudo = opts
+        .root
+        .as_ref()
+        .map(|inner| inner.as_ref().map(String::as_str));
```

Comment:
Can we use `as_deref()` here instead of the `as_ref().map(..)` construction?

Reply:
Sure thing, updated.